## Final Project Submission

Please fill out: 
* Student name: Willian Nyawir
* Student pace: self paced / part time / full time
* Scheduled project review date/time: Friday 17th Dec 2026
* Instructor name: Mrs Mwikali
* Blog post URL:


# ANALYSIS OF THE DATASET

# BUSINESS PROBLEM
The company plans to expand into the aviation industry but lacks knowledge about the safety risks associated with different aircraft. This project analyzes historical aviation accident data to identify aircraft makes and models with lower risk profiles, enabling data-driven decisions on which aircraft to purchase for commercial and private operations.


# DATA UNDERSTANDING

In [9]:
#Importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
df=pd.read_csv('Aviation_Data.csv')
df

C:\Users\user\AppData\Local\Temp\ipykernel_20772\2010058794.py:5: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv('Aviation_Data.csv')


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [11]:
df.describe() #gives statistical summary of numerical columns

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [12]:
df.columns #lists all column names

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [13]:
df.shape #calculates number of rows and columns

(90348, 31)

In [14]:
df.ndim #checks number of dimensions

2

The above data has 90348 rows and 31 columns. It also has 2 dimensions. The statistical report of the numerical data is also shown above

## DATA CLEANING

In [16]:
used_columns = df[['Make','Model','Injury.Severity','Aircraft.damage','Aircraft.Category',
                   'Number.of.Engines','Engine.Type','Purpose.of.flight',
                   'Weather.Condition','Total.Fatal.Injuries',
                   'Total.Serious.Injuries','Total.Minor.Injuries','Registration.Number']]

#It organizes the column titles by removing spaces, converting to lowercase and replacing periods with underscores
used_columns.columns = (
    used_columns.columns
    .str.strip()
    .str.lower()
    .str.replace('.', '_', regex=False)
)
used_columns

make      model injury_severity aircraft_damage  \
0                         Stinson      108-3        Fatal(2)       Destroyed   
1                           Piper   PA24-180        Fatal(4)       Destroyed   
2                          Cessna       172M        Fatal(3)       Destroyed   
3                        Rockwell        112        Fatal(2)       Destroyed   
4                          Cessna        501        Fatal(1)       Destroyed   
...                           ...        ...             ...             ...   
90343                       PIPER  PA-28-151           Minor             NaN   
90344                    BELLANCA       7ECA             NaN             NaN   
90345  AMERICAN CHAMPION AIRCRAFT      8GCBC       Non-Fatal     Substantial   
90346                      CESSNA       210N             NaN             NaN   
90347                       PIPER  PA-24-260           Minor             NaN   

      aircraft_category  number_of_engines    engine_type purpose_of_flight  \
0                   NaN                1.0  Reciprocating          Personal   
1                   NaN                1.0  Reciprocating          Personal   
2                   NaN                1.0  Reciprocating          Personal   
3                   NaN                1.0  Reciprocating          Personal   
4                   NaN                NaN            NaN          Personal   
...                 ...                ...            ...               ...   
90343               NaN                NaN            NaN          Personal   
90344               NaN                NaN            NaN               NaN   
90345          Airplane                1.0            NaN          Personal   
90346               NaN                NaN            NaN          Personal   
90347               NaN                NaN            NaN          Personal   

      weather_condition  total_fatal_injuries  total_serious_injuries  \
0                   UNK                   2.0                     0.0   
1                   UNK                   4.0                     0.0   
2                   IMC                   3.0                     NaN   
3                   IMC                   2.0                     0.0   
4                   VMC                   1.0                     2.0   
...                 ...                   ...                     ...   
90343               NaN                   0.0                     1.0   
90344               NaN                   0.0                     0.0   
90345               VMC                   0.0                     0.0   
90346               NaN                   0.0                     0.0   
90347               NaN                   0.0                     1.0   

       total_minor_injuries registration_number  
0                       0.0              NC6404  
1                       0.0              N5069P  
2                       NaN              N5142R  
3                       0.0              N1168J  
4                       NaN               N15NY  
...                     ...                 ...  
90343                   0.0              N1867H  
90344                   0.0              N2895Z  
90345                   0.0              N749PJ  
90346                   0.0              N210CU  
90347                   0.0              N9026P  

[90348 rows x 13 columns]

In [17]:
used_columns.isna().sum() #counts no. of null values


make                       1522
model                      1551
injury_severity            2459
aircraft_damage            4653
aircraft_category         58061
number_of_engines          7543
engine_type                8555
purpose_of_flight          7651
weather_condition          5951
total_fatal_injuries      12860
total_serious_injuries    13969
total_minor_injuries      13392
registration_number        2841
dtype: int64

In [21]:
used_columns=used_columns.dropna() #removes null values
used_columns

make   model injury_severity aircraft_damage  \
7                 Cessna     140       Non-Fatal     Substantial   
8                 Cessna    401B       Non-Fatal     Substantial   
12              Bellanca  17-30A       Non-Fatal       Destroyed   
13                Cessna   R172K        Fatal(1)       Destroyed   
14                Navion       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines    engine_type purpose_of_flight  \
7              Airplane                1.0  Reciprocating          Personal   
8              Airplane                2.0  Reciprocating          Business   
12             Airplane                1.0  Reciprocating          Personal   
13             Airplane                1.0  Reciprocating          Personal   
14             Airplane                1.0  Reciprocating          Personal   
...                 ...                ...            ...               ...   
90098          Airplane                1.0  Reciprocating          Personal   
90106          Airplane                1.0  Reciprocating          Personal   
90120          Airplane                2.0  Reciprocating          Personal   
90194          Airplane                1.0  Reciprocating              ASHO   
90226          Airplane                1.0  Reciprocating          Personal   

      weather_condition  total_fatal_injuries  total_serious_injuries  \
7                   VMC                   0.0                     0.0   
8                   IMC                   0.0                     0.0   
12                  IMC                   0.0                     0.0   
13                  IMC                   1.0                     0.0   
14                  IMC                   1.0                     0.0   
...                 ...                   ...                     ...   
90098               VMC                   0.0                     0.0   
90106               VMC                   0.0                     0.0   
90120               VMC                   0.0                     0.0   
90194               VMC                   0.0                     1.0   
90226               VMC                   0.0                     0.0   

       total_minor_injuries registration_number  
7                       0.0              N2482N  
8                       0.0              N7967Q  
12                      1.0              N14779  
13                      0.0              N758SK  
14                      0.0              N4876K  
...                     ...                 ...  
90098                   0.0              N18158  
90106                   0.0              N2047Q  
90120                   0.0              N2889W  
90194                   0.0              N778MS  
90226                   0.0              N1855K  

[20345 rows x 13 columns]

In [23]:
airplane_dataset=used_columns.loc[used_columns['aircraft_category']=='Airplane']
airplane_dataset

make   model injury_severity aircraft_damage  \
7                 Cessna     140       Non-Fatal     Substantial   
8                 Cessna    401B       Non-Fatal     Substantial   
12              Bellanca  17-30A       Non-Fatal       Destroyed   
13                Cessna   R172K        Fatal(1)       Destroyed   
14                Navion       A        Fatal(1)       Destroyed   
...                  ...     ...             ...             ...   
90098             CESSNA     150       Non-Fatal     Substantial   
90106             CESSNA   177RG       Non-Fatal     Substantial   
90120              BEECH    B-60       Non-Fatal     Substantial   
90194  STEPHEN J HOFFMAN  MS-500           Minor     Substantial   
90226           LUSCOMBE      8E       Non-Fatal     Substantial   

      aircraft_category  number_of_engines    engine_type purpose_of_flight  \
7              Airplane                1.0  Reciprocating          Personal   
8              Airplane                2.0  Reciprocating          Business   
12             Airplane                1.0  Reciprocating          Personal   
13             Airplane                1.0  Reciprocating          Personal   
14             Airplane                1.0  Reciprocating          Personal   
...                 ...                ...            ...               ...   
90098          Airplane                1.0  Reciprocating          Personal   
90106          Airplane                1.0  Reciprocating          Personal   
90120          Airplane                2.0  Reciprocating          Personal   
90194          Airplane                1.0  Reciprocating              ASHO   
90226          Airplane                1.0  Reciprocating          Personal   

      weather_condition  total_fatal_injuries  total_serious_injuries  \
7                   VMC                   0.0                     0.0   
8                   IMC                   0.0                     0.0   
12                  IMC                   0.0                     0.0   
13                  IMC                   1.0                     0.0   
14                  IMC                   1.0                     0.0   
...                 ...                   ...                     ...   
90098               VMC                   0.0                     0.0   
90106               VMC                   0.0                     0.0   
90120               VMC                   0.0                     0.0   
90194               VMC                   0.0                     1.0   
90226               VMC                   0.0                     0.0   

       total_minor_injuries registration_number  
7                       0.0              N2482N  
8                       0.0              N7967Q  
12                      1.0              N14779  
13                      0.0              N758SK  
14                      0.0              N4876K  
...                     ...                 ...  
90098                   0.0              N18158  
90106                   0.0              N2047Q  
90120                   0.0              N2889W  
90194                   0.0              N778MS  
90226                   0.0              N1855K  

[17886 rows x 13 columns]

# DATA PROCESSING